### DISCUSSION TITLE
Is it possible to add a switch for downloading subtitles?

### DISCUSSION THREAD

**Discussion by domin80r** [eppic/ytBATCH#1] [Ideas]
YouTube videos can have **subtitles** in multiple languages, is it possible to add a switch for subtitles to be downloaded in SRT (or VTT) format along with the video?

I know there can also be variations of the language dialects (eg "English", "English (United Kingdom)", "English (United States)", "English - Default", etc.) that would need to be supported somehow.

- Example with multiple language subtitles including "English": https://www.youtube.com/watch?v=0XUlS_2B5ZA
- Example with "English (United Kingdom)": https://www.youtube.com/watch?v=cxidCRyCQ7o
- Example with "English - Default": https://www.youtube.com/watch?v=VNOq5xYxsNU

Thanks, and I love ytBATCH, it makes it so easy to use. Nice work.

**Comment by eppic** [comment]
I will look into this. there even used to be a function for subtitles, however I removed it due to being unreliable. Sadly, I am restricted to the tools yt-dlp provides so subtitle support can only get that far.
However, Subtitles aren't that hard to download with yt-dlp:
You need to provide custom parameters in the "youtube-dl.conf" (`(P) Preferences... -> (V) Open yt-dlp Config... `) 

Option 1: 
Add `--write-subs`
This writes a `.vtt` file with the same name as the video

Option 2: 
Add `--embed-subs`
This writes the subtitles directly into the video, no extra file. However, this only works with .mp4, .webm and .mkv.

If you want to download subtitles for one language only, you will need to use `--sub-langs LANGS` and provide the code for your wanted language (Instead of `LANGS`) You can find the supported languages using `--list-subs`.

For more help Enter (H) in the Main Menu of ytBATCH. (Shortcut to yt-dlp --help)

Thank you for the nice words :) I'm glad you like ytBATCH. I sadly do not have the time to work on it right now, but I hope I was able to help anyway.

**Reply by domin80r** [reply]
Thanks @eppic , those options are great for me. I'll use the embed function and then choose my language in VLC or Kodi.
